Components/StockPage.js

```diff
@@ -100,7 +103,43 @@ export default function StockPage(props) {
     };
   }, []);
 
+  //trying to read favorites from DB on launch
+ 
+
+  const readFavoritesFromDB = () => {
+    let isMounted = true;
+    if (isMounted) {
+      let uid = firebase.auth().currentUser.uid;
+      let favoriteRef = database().ref(`${uid}/favorites/${companySymbol}`);
+      if(favoriteRef) {
+        //does get correct path 
+        console.log(favoriteRef);
+        favoriteRef.on('value', (snapshot) => {
+          console.log('Favorited on read from DB? ' + snapshot.val());
+          if(snapshot.val() === 'true' || snapshot.val() === 'false') {
+            setFavorited(snapshot.val());
+          }
+        });
+      } else {
+        console.log('favoriteRef is ' + favoriteRef.toString());
+      }
+    }
+
+    return () => {
+      isMounted = false;
+      console.log('isMounted FALSE');
+    };
+  };
+  
   useEffect(() => {
```

Comment:
why are we pushing or removing if favorited is true only? this would mean that it only pushes when true, so we can't set to false ever, right?